core/client/fs/src/main/java/alluxio/client/file/cache/DefaultMetaStore.java

```diff
@@ -36,6 +37,8 @@
   private static final Logger LOG = LoggerFactory.getLogger(DefaultMetaStore.class);
   /** A map from PageId to page info. */
   private final Map<PageId, PageInfo> mPageMap = new HashMap<>();
+  /** A map from fileId to file info.*/
+  private final Map<String, FileInfo> mFileMap = new WeakHashMap<>();
```

Comment:
> Using a weak map here would help kick out the fileInfo instances which are not associated with any pageInfo

The idea is good, but we found it had some problems in our environment. When the element in mPageMap is removed, the element in mFileMap related to this removed file will be removed when gc is happened. For example, mPageMap has 2 elements{(pageId(file 1, pageIndex 0), pageInfo),(pageId(file 1, pageIndex 1),pageInfo)}. mFileMap has 1 element{(file 1, fileInfo)}. When element(pageId(file 1, pageIndex 0), pageInfo)  is removed in mPageMap, the element{(file 1, fileInfo)} will be removed by gc. However, element(pageId(file 1, pageIndex 1),pageInfo) is still in mPageMap and no element associated with this page in FileMap. So we cannot get this page by get() and it will lead to read data again from remote. So we suggest use reference counting to solve it.